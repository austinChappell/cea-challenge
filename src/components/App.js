import React, { Component } from 'react';

import Api from '../api/api';

const api = new Api();

const { authorize, getMeetupData } = api;

class App extends Component {
  componentDidMount() {
    // getMeetupData('reactjs-dallas', this.printRes);
  }

  getData = () => {
    authorize('reactjs-dallas', this.printRes);
  }

  printRes = (results) => {
    console.log('RESPONSE', results)
  }

  render() {
    return (
      <div className="App">
        <a href={`https://secure.meetup.com/oauth2/authorize?client_id=${process.env.REACT_APP_MEETUP_CONSUMER_KEY}&response_type=token&redirect_uri=${process.env.REACT_APP_MEETUP_REDIRECT_URI}`}>Authorize</a>
      </div>
    );
  }
}

export default App;
