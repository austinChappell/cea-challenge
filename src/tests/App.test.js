import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import App from '../components/App';

describe('The App Component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<App />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
})
;