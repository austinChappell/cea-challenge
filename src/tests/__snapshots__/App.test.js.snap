// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The App Component should render correctly 1`] = `
<div
  className="App"
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <BrowserRouter>
      <Switch>
        <Route
          component={[Function]}
          path="/group/:groupName/event/:eventId"
        />
        <Route
          component={[Function]}
          path="/group/:groupName"
        />
        <Route
          component={[Function]}
          path="/"
        />
      </Switch>
    </BrowserRouter>
  </Provider>
</div>
`;
